test/test-genesis.ts

```diff
@@ -31,66 +53,108 @@ describe("Genesis Contract", () => {
   });
 
   // TODO Adjust with real values
-  it("Should initialize the Genesis contract", async () => {
+  it('Should initialize the Genesis contract', async () => {
     expect(await contract.MAX_SUPPLY()).to.equal(1000);
-    expect(await contract.PRICE()).to.equal(utils.parseEther("0.0000001"));
+    expect(await contract.PRICE()).to.equal(utils.parseEther('0.0000001'));
     expect(await contract.MAX_PER_MINT()).to.equal(1);
-    expect(await contract.presaleActive()).to.be.false;
     expect(await contract.mintActive()).to.be.false;
-    expect(await contract.reservesMinted()).to.be.false;
   });
 
-  it("Should set the right owner", async () => {
+  it('Should set the right owner', async () => {
     expect(await contract.owner()).to.equal(await owner.address);
   });
 
-  it("Should not allow minting if it is not active", async function () {
-    await expect(contract.connect(address1).mint()).to.be.revertedWith(
-      "Minting is not active yet!",
-    );
-    await expect(contract.connect(owner).mintNFTs(5)).to.be.revertedWith(
-      "Minting is not active yet!",
-    );
+  it('Should not allow minting if it is not active', async function () {
+    await expect(
+      mint(
+        owner,
+        bogusNonce,
+        bogusProof,
```

Comment:
What if the proper proof are sent? Maybe write a comment if you send the bogus because of the order of the requires in the contract. MINOR.

Reply:
I don't get this

Reply:
but I think it's resolved since I switched to redeploying the contract for every tests

Reply:
I meant, why do you get an error saying mint is not active although you send it invalid data.
Ok if tests are now independent.